src/components/customerPortal/PortalInvoicesList.tsx

```diff
@@ -68,10 +76,50 @@ gql`
     }
   }
 
+  query getCustomerPortalInvoicesCollection($expireCache: Boolean) {
+    customerPortalInvoiceCollections(expireCache: $expireCache) {
+      collection {
+        amountCents
+        invoicesCount
+        currency
+      }
+    }
+  }
+
+  query getCustomerPortalOverdueBalances($expireCache: Boolean) {
+    customerPortalOverdueBalances(expireCache: $expireCache) {
+      collection {
+        amountCents
+        currency
+        lagoInvoiceIds
+      }
+    }
+  }
+
+  query getCustomerPortalUserCurrency {
```

Comment:
About that `customerPortalUser` query. 
I can see it's performed twice in your PR, and probably more in the current code or next to come. (just spotted also in `src/components/customerPortal/PortalCustomerInfos.tsx`)

That sound like a data we may want to share across all the customer portal and that won't necessarily change once we load the page.
Did you consider putting that customer fetch logic in a reusable hook, method or any place that could allow fetching those informations only once and re-use it elsewhere?

Reply:
Good suggestion, but currently each query has some logic embedded below (the loading / error states etc), so I would keep this as a future improvement (they fetch different data anyways, but your solution would indeed reduce the number of calls).

Reply:
ok! Yeah I think it would be a real benefit reducing some api calls as we're adding more and more around